src/python/pants/backend/docker/subsystems/dockerfile_parser_test.py

```diff
@@ -107,6 +107,25 @@ def test_build_args(rule_runner: RuleRunner) -> None:
     )
 
 
+def test_from_image_build_arg_names(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "test/a/BUILD": """docker_image(name="image")""",
```

Comment:
Also what about labeling these directories `upstream` and `downstream` rather than `a` and `b`? I think it will make this clearer.